src/common/intrusive_lru.h

```diff
@@ -122,9 +122,9 @@ class intrusive_lru {
 
   // when the lru_set exceeds its target size, evict
   // only unreferenced elements from it (if any).
-  void evict() {
+  void evict(unsigned target_size) {
```

Comment:
We can either set a default value here of `lru_target_size` instead of adding it to all the trivial callers. Moreover, since there are only 2 possible target_size (target_size or zero) we could use a boolean instead.

```
 void evict(bool force_full_evict = false) {
   ...
   !force_full_evict && lru_set.size() > lru_target_size
```

Reply:
I really prefer simply passing in the target.  A boolean simply requires the reader to check somewhere else to determine the behavior.  I'm not a huge fan of default arguments for small helpers like this as it makes the behavior less obvious.